src/setups/complex_shape.jl

```diff
@@ -41,13 +41,25 @@ For more information about the method see [`WindingNumberJacobson`](@ref) or [`W
 !!! warning "Experimental Implementation"
     This is an experimental feature and may change in any future releases.
 """
-function ComplexShape(geometry::Union{TriangleMesh, Polygon}; particle_spacing, density,
+struct ComplexShape{S, IC, ICB, SDF, IG, WN, ELTYPE}
```

Comment:
Why is this a struct now? Especially since the packing system can also accept a plain `InitialCondition`.

Reply:
I'm not really happy with it yet either. I'm not really sure about the API. I think we should this discuss first. 
The issue is, that this holds also the boundary IC with the same initial particle grid and the same SDF without generating another SDF.